src/components/SearchBar.vue

```diff
@@ -1,7 +1,15 @@
 <script setup lang="ts">
-import { ref } from 'vue'
 import searchIcon from '../assets/icons/Vector.svg'
-const searchQuery = ref('')
+
+const emit = defineEmits<{
+  (e: 'search', query: string): void
+}>()
+
+const searchQuery = defineModel<string>()
```

Comment:
This can be restructured to provide more stability


```suggestion
const searchQuery = defineModel({
  required: true, // optional
  default: '' // will make the model of type string implicitly 
})

defineEmits<{
  search: [value: typeof searchQuery['value']] // now emitted value type is coupled to the definition of model
}>()
```